### DISCUSSION TITLE
Understand the role of the binding better

### DISCUSSION THREAD

**Discussion by LoyalPotato** [davidbyttow/govips#418] [Q&A] [answered]
Hello,

I wanted to understand if the binding is 1:1 with the functions of libvips, which I don't think it is, since there are some options missing? My reasoning for this is this:

The Python bindings have this section for `ForeignHeifCompression` used in [`heifsave_buffer`](https://libvips.github.io/pyvips/vimage.html?highlight=heif#pyvips.Image.heifsave_buffer)

The same is true for the C bindings (if it's not the correct term, apologies), [`vips_heif_save_buffer`](https://www.libvips.org/API/current/VipsForeignSave.html#vips-heifsave-buffer).

I couldn't find this in `govips`. So I wanted to ask if it's not a 1:1, why?

Thank you :)

**Comment by tonimelisma** [comment, marked as answer]
Hey @LoyalPotato, no it's not 1:1 because not all of the libvips functionality has been implemented. It's a question of time and effort. If there's missing functionality I do try to review and merge pull requests ASAP, so you're more than happy to contribute whatever you need!

**Reply by LoyalPotato** [reply]
Oh got it👍 

Thank you for the fast response!
